share/spack/qa/setup-env-test.sh

```diff
@@ -200,3 +200,6 @@ despacktivate
 echo "Correct error exit codes for activate and deactivate"
 fails spack env activate nonexisiting_environment
 fails spack env deactivate
+
+echo "Correct error exit codes for unit-test when it fails"
+fails spack unit-test fail
```

Comment:
@alalazo  This seems like a good way to test that command return statements translate to exit codes, nice!  I was unaware of this file at all, but now that I'm looking through it, it seems like the natural place for this test.